tests/conftest.py

```diff
@@ -225,3 +226,32 @@ def variants_conda_build_sysroot(monkeypatch, request):
         ).stdout.strip(),
     )
     return request.param
+
+
+# see https://github.com/pytest-dev/pytest/issues/363#issuecomment-1335631998
+@pytest.fixture(scope="session")
+def monkeysession() -> pytest.MonkeyPatch:
+    with pytest.MonkeyPatch.context() as mp:
+        yield mp
+
+
+@pytest.fixture(scope="session")
+def conda_build_test_recipe_path(
+    tmp_path_factory: pytest.TempPathFactory,
+    monkeysession: pytest.MonkeyPatch,
+) -> Path:
+    """Clone conda_build_test_recipe.
+
+    This exposes the special dummy package "source code" used to test various git/svn/local recipe configurations.
+    """
+    # clone conda_build_test_recipe locally
+    repo = tmp_path_factory.mktemp("conda_build_test_recipe", numbered=False)
+    subprocess.run(
+        ["git", "clone", "https://github.com/conda/conda_build_test_recipe", str(repo)],
+        check=True,
+    )
+
+    # provide cloned repo as envvar
+    monkeysession.setenv("CONDA_BUILD_TEST_RECIPE_PATH", repo)
```

Comment:
This is only an environment variable that makes itself visible to the whole session?

It makes me nervous to see monkey patching that affects all tests.

For example, if a test uses the environment variable, forgets to ask for the fixture, but happens to run after the first test that *does* ask for the fixture - have we written an order-dependent test?

Reply:
This a valid concern, that's why I explicitly checked that the tests passed as `fixture(scope="function")`.

This is an ok risk IMO since we are unlikely to write more tests with this special repo. If anything, we'll be looking to remove the need for this special repo altogether so this is just a transitionary fixture.

Reply:
Split fixture into `conda_build_test_recipe_path` and `conda_build_test_recipe_envvar`